src/controller/AbstractMdnsDiscoveryController.cpp

```diff
@@ -31,6 +31,8 @@ namespace Controller {
 
 void AbstractMdnsDiscoveryController::OnNodeDiscoveryComplete(const chip::Mdns::DiscoveredNodeData & nodeData)
 {
+    chip::Protocols::UserDirectedCommissioning::UserDirectedCommissioningServer::GetInstance().OnCommissionableNodeFound(nodeData);
```

Comment:
We need to unwind what's happening with these singletons.  Here's what's evolving:

* `chip::Mdns::Resolver` is a singleton
* Why?  This might make sense for minimal mDNS, but makes no sense for Avahi or Bonjour
* The singleton pattern then propagates into `AbstractMdnsDiscoveryController`
* -> even though this does not outwardly appear to be a singleton, we cannot instantiate more than one due to shared state
* Now `AbstractMdnsDiscoveryController` is calling into the UDC Server singleton
* -> at this point, we are even further from being able to instantiate more than one `AbstractMdnsDiscoveryController`
* and because `AbstractMdnsDiscoveryController` is a member of `chip::Controller::DeviceController` that then is further hostage to the singleton pattern as well

The thing is, `UserDirectedCommissioningServer` is actually a reasonable candidate for the singleton pattern.  There isn't any particularly compelling reason for a device to provide multiple UDC services.  However, we have a looming problem in the resolver code.  There is zero reason we should have this as a singleton in unconstrained platforms.

So while this PR doesn't create this problem, it is making it worse.  Can we provide a different mechanism for `AbstractMdnsDiscoveryController` to notify `UserDirectedCommissioningServer` that isn't this inline reference to the UDC server instance?  This would prevent this new singleton coupling and make it easier for us to address the fact the `AbstractMdnsDiscoveryController` is also effectively a singleton, which makes no sense.

Reply:
Another observation: this feature also seems itself to be a victim of the singleton pattern that's developing.  Won't the UDC server break if a request comes in while the resolver singleton is in use?  Doesn't the UDC server really just need its own resolver?

Reply:
Removed the singleton pattern so marking this comment as resolved